tests/expected/uninit/fixme_unions.rs

```diff
@@ -48,14 +47,59 @@ unsafe fn union_as_subfields_should_pass() {
     let padding = u1.a; // Read 2 bytes from `u`.
 }
 
+/// Tests initialized access if unions are top-level subfields.
+#[kani::proof]
+unsafe fn union_as_subfields_should_fail() {
+    let u = U { a: 0 }; // `u` is initialized for 2 bytes.
+    let s = S { u };
+    let s1 = s;
+    let u1 = s1.u; // `u1` is initialized for 2 bytes.
+    let padding = u1.b; // Read 4 bytes from `u`.
+}
+
 union Outer {
     u: U,
     a: u32,
 }
 
-/// Tests unions composing with other unions.
+/// Tests unions composing with other unions and reading non-padding data.
 #[kani::proof]
 unsafe fn uber_union_should_pass() {
     let u = Outer { u: U { b: 0 } }; // `u` is initialized for 4 bytes.
     let non_padding = u.a; // Read 4 bytes from `u`.
 }
+
+/// Tests unions composing with other unions and reading padding data.
```

Comment:
This is an `expected` test, right? Don't we need to add an expectation that will fail if we fix this issue?

I would also prefer having multiple fixme tests, one per limitation.

Reply:
I wasn't sure what an appropriate failing behavior would be, so I left it as a TODO comment for now.